controller/volume_controller.go

```diff
@@ -1564,11 +1564,14 @@ func (c *VolumeController) reconcileVolumeCondition(v *longhorn.Volume, e *longh
 			continue
 		}
 		if v.Spec.DataLocality == longhorn.DataLocalityStrictLocal {
-			if v.Spec.NodeID == "" {
-				continue
+			if v.Status.RestoreRequired {
+				r.Spec.HardNodeAffinity = v.Status.OwnerID
```

Comment:
Can you explain where `v.Status.OwnerID` will be set when the volume is restoring? so it can be used here.

Reply:
For a volume being restored, the `volume.spec.nodeID` and `volume.status.ownerID` are empty, and the later one is determined in https://github.com/longhorn/longhorn-manager/blob/master/controller/volume_controller.go#L295.